### DISCUSSION TITLE
painutils/print on new multishell or shell tab

### DISCUSSION THREAD

**Discussion by DenavDot** [cc-tweaked/CC-Tweaked#1388] [Q&A] [answered]
So,whenever I try to use `print()` `printError()` `paintutils.X` after I've made and switched to a multishell or shell tab, the text/drawing still shows up on the original terminal, and not the new one. Am I doing something wrong?

here's a simple demonstration:
(P.S I open "lua" for the program,but I just need a basic screen to be able to draw/print in(like a monitor), so any program would do)
```lua
local t=shell.openTab("lua")
shell.switchTab(t)

print("testing!")
paintutils.drawFilledBox(5,5,10,10,colors.gray)

```

**Comment by Wojbie** [comment, marked as answer]
When you switchTab you only change which tab of multishell is active/visible to user. You code will keep executing in tab you ran it.
There is currently no way to just have code jump between tabs like this. Best you can do is have 2 programs communicate over events for example.

**Reply by DenavDot** [reply]
Thanks! I see what you mean,I'll try to work this out somehow
